source/extensions/filters/http/transformation/transformation_filter_config.cc

```diff
@@ -39,6 +40,77 @@ TransformerConstSharedPtr Transformation::getTransformer(
   }
 }
 
+OnStreamCompleteTransformerConstSharedPtr OnStreamCompleteTransformation::getTransformer(
+    const envoy::api::v2::filter::http::Transformation &transformation,
+    Server::Configuration::CommonFactoryContext &context) {
+  switch (transformation.transformation_type_case()) {
+  case envoy::api::v2::filter::http::Transformation::kTransformationTemplate:
+    return std::make_unique<InjaOnStreamCompleteTransformer>(
+        transformation.transformation_template());
+  case envoy::api::v2::filter::http::Transformation::kHeaderBodyTransform: {
+    throw EnvoyException("invalid transformer configured on onstreamcomplete: HeaderBodyTransform");
+  }
+  case envoy::api::v2::filter::http::Transformation::kTransformerConfig: {
+    auto &factory = Config::Utility::getAndCheckFactory<OnStreamCompleteTransformerExtensionFactory>(transformation.transformer_config());
+    auto config = Config::Utility::translateAnyToFactoryConfig(transformation.transformer_config().typed_config(), context.messageValidationContext().staticValidationVisitor(), factory);
+    return factory.createOnStreamCompleteTransformer(*config, context);
+  }
+  case envoy::api::v2::filter::http::Transformation::
+      TRANSFORMATION_TYPE_NOT_SET:
+    // TODO: return null here?
```

Comment:
I think we should remove the TODO and add trace logs if that's easy

Reply:
changed x4 [8e3a137](https://github.com/solo-io/envoy-gloo/pull/204/commits/8e3a137861f451dea9a888c00cc9f9e2712b8a14)